jolt-core/src/poly/rlc_polynomial.rs

```diff
@@ -70,41 +100,90 @@ impl<F: JoltField> RLCPolynomial<F> {
         }
     }
 
-    /// O(sqrt(T)) space Vector matrix product constructor for all polys
-    /// Allows for just a single pass over the trace
-    pub fn new_streaming(
-        dense_polys: Vec<(CommittedPolynomial, F)>,
-        onehot_polys: Vec<(CommittedPolynomial, F)>,
-        lazy_trace: LazyTraceIterator,
-        preprocessing: Arc<RLCStreamingData>,
-        one_hot_params: &OneHotParams,
+    /// Constructs an `RLCPolynomial` as a linear combination of `polynomials` with the provided
+    /// `coefficients`.
+    ///
+    /// This is a legacy helper (used by some commitment backends) that eagerly combines dense
+    /// polynomials into `dense_rlc` and stores one-hot polynomials lazily in `one_hot_rlc`.
+    #[allow(unused_variables)]
+    pub fn linear_combination(
+        poly_ids: Vec<CommittedPolynomial>,
+        polynomials: Vec<Arc<MultilinearPolynomial<F>>>,
+        coefficients: &[F],
+        streaming_context: Option<Arc<StreamingRLCContext<F>>>,
     ) -> Self {
+        debug_assert_eq!(polynomials.len(), coefficients.len());
+        debug_assert_eq!(polynomials.len(), poly_ids.len());
+
+        // Collect indices of dense (non-one-hot) polynomials.
```

Comment:
nit: use [partition](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.partition) to divide `polynomials` into dense vs one-hot